### PR TITLE
Implement DirectMeter extern for eBPF backend

### PR DESCRIPTION
**Pull request by kmateuszssak**
This PR adds DirectMeter extern.

Co-authored-by: Tomasz Osiński <tomasz@opennetworking.org>
Co-authored-by: Jan Palimąka <jan.palimaka@orange.com>

### CODE REVIEW

**Comment by kmateuszssak** [comment]
@mbudiu-vmw I will improve readability of meters PTF tests later. For now, pleas review as it is.

File: backends/ebpf/tests/p4testdata/meters-direct-and-counter.p4 [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <core.p4>
+#include <psa.p4>
+
+typedef bit<48>  EthernetAddress;
```
Comment (mbudiu-vmw) [inline backends/ebpf/tests/p4testdata/meters-direct-and-counter.p4:21]: I thought you were using some common definitions of the headers using an #include.
Forgot to check this in a few previous PRs, perhaps you want to scan the recent test additions.

File: backends/ebpf/tests/p4testdata/meters-direct-and-indirect.p4 [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+control egress(inout headers hdr,
+               inout metadata user_meta,
+               in    psa_egress_input_metadata_t  istd,
+               inout psa_egress_output_metadata_t ostd)
+{
+    apply { }
+}
+
+control CommonDeparserImpl(packet_out packet,
```
Comment (mbudiu-vmw) [inline backends/ebpf/tests/p4testdata/meters-direct-and-indirect.p4:145]: I am even wondering whether you couldn't also put all these controls into the common header file... would make the code much easier to read.

**Review by mbudiu-vmw — CHANGES_REQUESTED** [review]

**Comment by kmateuszssak** [comment]
@mbudiu-vmw please review.

File: backends/ebpf/tests/ptf/meters.py [hunk 3]
```diff
@@ -40,298 +135,190 @@ def runTest(self):
                           pir=250000, pbs=2500, cir=250000, cbs=2500)
         testutils.send_packet(self, PORT0, pkt)
         testutils.verify_packet(self, pkt, PORT1)
+
         # Expecting pbs_left, cbs_left 2500 B - 100 B = 2400 B -> 09 60
+        meter_value = build_meter_value(pir=250000, cir=250000, pbs=2500,
```
Comment (mbudiu-vmw) [inline backends/ebpf/tests/ptf/meters.py:140]: much better, isn't it?

**Review by mbudiu-vmw — APPROVED** [review]

**Pull request merged**